File: algorithms/1_easy/S/01796_second_largest_digit_in_a_string/solution.cpp
class Solution {
public:
    int secondHighest(string s) {
    
        array<int, 10> is_seen{};

        for (const char &ch: s) {
            if (ch >= '0' && ch <= '9') {
                is_seen[static_cast<size_t>(ch - '0')] = true;
            } 
        }

        int count{0};

        for (size_t i{9}; i < 10; --i) {
            if (is_seen[i]) {
                count += 1;

                if (count == 2) {
                    return i;
                }
            }
        }

        return -1;
    }
};

File: algorithms/1_easy/S/01796_second_largest_digit_in_a_string/solution.rs
impl Solution {
    pub fn second_highest(s: String) -> i32 {

        let mut is_seen = [false; 10];

        for byte in s.into_bytes() {
            if byte >= b'0' && byte <= b'9' {
                is_seen[(byte - b'0') as usize] = true;
            } 
        }

        let mut count = 0;

        for i in (0..=9).rev() {
            if is_seen[i] {
                count += 1;
                if count == 2 {
                    return i as i32;
                }
            }
        }

        -1
    }
}
